isis/src/control/apps/cnetedit/cnetedit.cpp

```diff
@@ -0,0 +1,1127 @@
+find files of those names at the top level of this repository. **/
+
+/* SPDX-License-Identifier: CC0-1.0 */
+
+#include "cnetedit.h"
+
+#include <QFile>
+#include <QList>
+#include <QMap>
+#include <QSet>
+#include <QString>
+#include <QTextStream>
+
+#include "ControlMeasure.h"
+#include "ControlNet.h"
+#include "ControlNetValidMeasure.h"
+#include "ControlPoint.h"
+#include "ControlPointList.h"
+#include "Cube.h"
+#include "FileName.h"
+#include "MeasureValidationResults.h"
+#include "Progress.h"
+#include "Pvl.h"
+#include "PvlGroup.h"
+#include "PvlKeyword.h"
+#include "PvlObject.h"
+#include "SerialNumber.h"
+#include "SerialNumberList.h"
+#include "IException.h"
+
+using namespace std;
+
+namespace Isis {
+  
+  // Deletion test
+  bool shouldDelete(ControlPoint *point);
+
+  // Mutator methods
+  void ignorePoint(ControlNet &cnet, ControlPoint *point, QString cause);
+  void ignoreMeasure(ControlNet &cnet, ControlPoint *point,
+                     ControlMeasure *measure, QString cause);
+  void deletePoint(ControlNet &cnet, int cp);
+  void deleteMeasure(ControlPoint *point, int cm);
+
+  // Edit passes
+  void populateLog(ControlNet &cnet, bool ignore);
+
+  void unlockPoints(ControlNet &cnet, ControlPointList &cpList);
+  void ignorePoints(ControlNet &cnet, ControlPointList &cpList);
+  void lockPoints(ControlNet &cnet, ControlPointList &cpList);
+
+  void unlockCubes(ControlNet &cnet, SerialNumberList &snl);
+  void ignoreCubes(ControlNet &cnet, SerialNumberList &snl);
+  void lockCubes(ControlNet &cnet, SerialNumberList &snl);
+
+  void unlockMeasures(ControlNet &cnet,
+      QMap< QString, QSet<QString> * > &editMeasures);
+  void ignoreMeasures(ControlNet &cnet,
+      QMap< QString, QSet<QString> * > &editMeasures);
+  void lockMeasures(ControlNet &cnet,
+      QMap< QString, QSet<QString> * > &editMeasures);
+
+  void checkAllMeasureValidity(ControlNet &cnet, SerialNumberList *serialNumbers);
+
+  // Validity test
+  MeasureValidationResults validateMeasure(const ControlMeasure *measure,
+      Cube *cube, Camera *camera);
+
+  // Logging helpers
+  void logResult(QMap<QString, QString> *pointsLog, QString pointId, QString cause);
+  void logResult(QMap<QString, PvlGroup> *measuresLog,
+      QString pointId, QString serial, QString cause);
+  PvlObject createLog(QString label, QMap<QString, QString> *pointsMap);
+  PvlObject createLog(QString label,
+      QMap<QString, QString> *pointsMap, QMap<QString, PvlGroup> *measuresMap);
+
+  // Global variables
+  int numPointsDeleted;
+  int numMeasuresDeleted;
+
+  bool deleteIgnored;
+  bool preservePoints;
+  bool retainRef;
+  bool keepLog;
+  bool ignoreAll;
+
+  QMap<QString, QString> *ignoredPoints;
+  QMap<QString, PvlGroup> *ignoredMeasures;
+  QMap<QString, QString> *retainedReferences;
+  QMap<QString, QString> *editLockedPoints;
+  QMap<QString, PvlGroup> *editLockedMeasures;
+
+  ControlNetValidMeasure *cneteditValidator;
```

Comment:
Is there any way to refactor this to not need global variables? These symbols could cause issues when linking libisis. Otherwise, we can make them static so they are scoped only to this .cpp file.

Reply:
Looking at this.

Reply:
Decided to go the static route unless there are objections.

Reply:
yeah, if it's not trivial to refactor. Static is fine.